doc/howto/storage_resize.md

```diff
@@ -0,0 +1,66 @@
+---
+discourse: 1333
+---
+
+# How to resize storage
+
+If you need more storage, you can increase the size of your storage pool or your storage volume.
+In some cases, it is also possible to reduce the size of a storage volume.
+
+## Grow a storage pool
+
+To increase the size of a storage pool, follow these general steps:
+
+1. Grow the size of the storage on disk.
+1. Let the file system know of the size change.
+1. Update the pool size configuration in LXD.
+
+See the specific commands for different storage drivers below.
+
+````{tabs}
+
+```{group-tab} Btrfs
+
+Enter the following commands to grow a loop-backed Btrfs pool by 5 G:
+
+    sudo truncate -s +5G <LXD_lib_dir>/disks/<pool_name>.img
+    sudo losetup -c <loop_device>
+    sudo btrfs filesystem resize max <LXD_lib_dir>/storage-pools/<pool_name>/
+    lxc storage set <pool_name> size <new_size>
```

Comment:
Is that command correct? It wasn't in the original instructions, but that should be the step to let LXD know about the change - right?

Reply:
Have you tried it? I'm wondering if LXD may reject modifying the setting as it knows it cannot apply the change (and won't know its been done outside of it).

Reply:
It works, but I'm not sure if it is actually needed, or if it would work without.
I suspect it would also work without, but then the general steps above (which I got from @stgraber ) aren't really correct.

Reply:
Its not technically needed, as LXD doesn't support resizing itself, so it never uses that value after initial creation.
However its nice to have the new size recorded just to avoid confusion in the future.

Reply:
Good to know. Although arguably if we are planning to implement resize pool support via changes to that key in the future, then keeping it in sync with the actual backend size now would make sense so that changes to that key in the future (once LXD has implemented pool resize support) correctly detects the new size.

Otherwise if, say it started at 10GiB, was then manually resize to 20GiB, but left as 10GiB in the `size` key, then if it was later changed to 20GiB LXD would try and resize to 20GiB (even though its already that size). Hopefully the implementations for each pool would cope with that though.

The docs would need to be updated anyway once pool resize is implemented.

Reply:
OK, then the actual general steps at the moment are only:

1. Grow the size of the storage on disk.
2. Let the file system know of the size change.
~~3. Update the pool size configuration in LXD.~~

Correct?

Reply:
Yep :)